flake.nix

```diff
@@ -214,6 +214,60 @@
                 # dev builds we do in nix shell, it just causes warning spew.
                 hardeningDisable = [ "all" ];
               };
+
+          # Overlay to override AFL++ package configuration. The default
+          # package is based on Clang14, which has some issues when building
+          # the C++20 code. The manual substitution of the postPatch is
+          # required because the package hard-codes this clang dependency.
+          aflplusplus = pkgs.aflplusplus.overrideAttrs {
+            nativeBuildInputs = [
+              pkgs.makeWrapper
+              pkgs.which
+              pkgs.clang_18
+              pkgs.gcc
+            ];
+            buildInputs = [
+              pkgs.llvmPackages_18.llvm
+              pkgs.llvmPackages_18.bintools
+              pkgs.gmp
+              pkgs.python3
+            ];
+            postPatch = ''
+              # See the original file [1]. This will need to be updated
+              # when this is changed, but maybe a better solution can be
+              # found. Comments are ommited here for brevity, only the
+              # clang packages are changed.
+              # [1] https://github.com/NixOS/nixpkgs/blob/release-24.11/pkgs/tools/security/aflplusplus/default.nix
+              rm Android.bp
+              substituteInPlace src/afl-cc.c \
+                --replace-fail "CLANGPP_BIN" '"${pkgs.clang_18}/bin/clang++"' \
+                --replace-fail "CLANG_BIN" '"${pkgs.clang_18}/bin/clang"' \
+                --replace-fail '"gcc"' '"${pkgs.gcc}/bin/gcc"' \
+                --replace-fail '"g++"' '"${pkgs.gcc}/bin/g++"' \
+                --replace-fail 'getenv("AFL_PATH")' "(getenv(\"AFL_PATH\") ? getenv(\"AFL_PATH\") : \"$out/lib/afl\")"
+              substituteInPlace src/afl-ld-lto.c \
+                --replace-fail 'LLVM_BINDIR' '"/nixpkgs-patched-does-not-exist"'
+              sed -i 's|LLVM_BINDIR = .*|LLVM_BINDIR = |' utils/aflpp_driver/GNUmakefile
+              substituteInPlace utils/aflpp_driver/GNUmakefile \
+                --replace-fail 'LLVM_BINDIR = ' 'LLVM_BINDIR = ${pkgs.clang_18}/bin/'
+              substituteInPlace GNUmakefile.llvm \
+                --replace-fail "\$(LLVM_BINDIR)/clang" "${pkgs.clang_18}/bin/clang"
+            '';
+          };
```

Comment:
Was there an issue with https://pastes.dxuuu.xyz/x5zobb ?

Reply:
I honestly couldn't remember and was trying to page back in some details. I've taken a pass at this, and it should now be easy to tie together the LLVM versions. It seems to work (let's see if CI passes!).

Reply:
I'm happy with this now. It's pretty straight-forward but isn't hard-coded to the specific version. (Although that pinning still occurs, it's at the top-level with the single environment definition.)